src/SDK/SmartDetectorsSDK/Alert.cs

```diff
@@ -19,7 +19,8 @@ public abstract class Alert
         /// </summary>
         /// <param name="title">The Alert's title.</param>
         /// <param name="resourceIdentifier">The resource identifier that this Alert applies to.</param>
-        protected Alert(string title, ResourceIdentifier resourceIdentifier)
+        /// <param name="occurenceTime">The exact time at which the issue that caused the alert has occured. If this is a continuous issue - pass the issue start time.</param>
+        protected Alert(string title, ResourceIdentifier resourceIdentifier, DateTime occurenceTime)
```

Comment:
>DateTime occurenceTime) [](start = 77, length = 23)

please add an overload for the ctor which defaults the occurrence time to DateTime.UtcNow

and revert all changed you've made to the project template

Reply:
and to the test code

---
In reply to: [244732284](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/44#discussion_r244732284) [](ancestors = 244732284)